src/platform/web/ui/general/ListView.ts

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2020 Bruno Windels <bruno@windels.cloud>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {el} from "./html.js";
+import {mountView, insertAt} from "./utils";
+import {BaseObservableList as ObservableList} from "../../../../observable/list/BaseObservableList.js";
+import {UIView, IMountArgs} from "./types";
+
+interface IOptions<T, V> {
+    list: ObservableList<T>,
+    onItemClick?: (childView: V, evt: UIEvent) => void,
+    className?: string,
+    tagName?: string,
+    parentProvidesUpdates?: boolean
+}
+
+type SubscriptionHandle = () => undefined;
+
+export class ListView<T, V extends UIView> implements UIView {
+
+    private _onItemClick?: (childView: V, evt: UIEvent) => void;
+    private _list: ObservableList<T>;
+    private _className?: string;
+    private _tagName: string;
+    private _root?: HTMLElement;
+    private _subscription?: SubscriptionHandle;
+    private _childCreator: (value: T) => V;
+    private _childInstances?: V[];
+    private _mountArgs: IMountArgs;
+
+    constructor(
+        {list, onItemClick, className, tagName = "ul", parentProvidesUpdates = true}: IOptions<T, V>, 
+        childCreator: (value: T) => V
+    ) {
+        this._onItemClick = onItemClick;
+        this._list = list;
+        this._className = className;
+        this._tagName = tagName;
+        this._root = undefined;
+        this._subscription = undefined;
+        this._childCreator = childCreator;
+        this._childInstances = undefined;
+        this._mountArgs = {parentProvidesUpdates};
+    }
+
+    root(): HTMLElement {
+        // won't be undefined when called between mount and unmount
+        return this._root!;
+    }
+
+    update(attributes: IOptions<T, V>) {
+        if (attributes.list) {
+            if (this._subscription) {
+                this._unloadList();
+                while (this._root!.lastChild) {
+                    this._root!.lastChild.remove();
+                }
+            }
+            this._list = attributes.list;
+            this.loadList();
+        }
+    }
+
+    mount(): HTMLElement {
+        const attr: {[name: string]: any} = {};
+        if (this._className) {
+            attr.className = this._className;
+        }
+        this._root = el(this._tagName, attr);
+        this.loadList();
+        if (this._onItemClick) {
+            this._root!.addEventListener("click", this);
+        }
+        return this._root!;
```

Comment:
And this,
```suggestion
        return root;
```
In short, no `!` needed?

Reply:
Have done so in https://github.com/vector-im/hydrogen-web/commit/b71a26b04a6368921279fd52d26e0a0b21d47e73 in #486